subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/configurations/DefaultConfigurationFactory.java

```diff
@@ -0,0 +1,142 @@
+    private final ImmutableAttributesFactory attributesFactory;
+    private final DocumentationRegistry documentationRegistry;
+    private final UserCodeApplicationContext userCodeApplicationContext;
+    private final DomainObjectContext owner;
+    private final ProjectStateRegistry projectStateRegistry;
+    private final WorkerThreadRegistry workerThreadRegistry;
+    private final DomainObjectCollectionFactory domainObjectCollectionFactory;
+    private final CalculatedValueContainerFactory calculatedValueContainerFactory;
+
+    @Inject
+    public DefaultConfigurationFactory(
+        Instantiator instantiator,
+        ConfigurationResolver resolver,
+        ListenerManager listenerManager,
+        DependencyMetaDataProvider metaDataProvider,
+        DomainObjectContext domainObjectContext,
+        FileCollectionFactory fileCollectionFactory,
+        BuildOperationExecutor buildOperationExecutor,
+        PublishArtifactNotationParserFactory artifactNotationParserFactory,
+        ImmutableAttributesFactory attributesFactory,
+        DocumentationRegistry documentationRegistry,
+        UserCodeApplicationContext userCodeApplicationContext,
+        DomainObjectContext owner,
+        ProjectStateRegistry projectStateRegistry,
+        WorkerThreadRegistry workerThreadRegistry,
+        DomainObjectCollectionFactory domainObjectCollectionFactory,
+        CalculatedValueContainerFactory calculatedValueContainerFactory
+    ) {
+        this.instantiator = instantiator;
+        this.resolver = resolver;
+        this.listenerManager = listenerManager;
+        this.metaDataProvider = metaDataProvider;
+        this.domainObjectContext = domainObjectContext;
+        this.fileCollectionFactory = fileCollectionFactory;
+        this.buildOperationExecutor = buildOperationExecutor;
+        this.artifactNotationParser = artifactNotationParserFactory.create();
+        this.capabilityNotationParser = new CapabilityNotationParserFactory(true).create();
+        this.attributesFactory = attributesFactory;
+        this.documentationRegistry = documentationRegistry;
+        this.userCodeApplicationContext = userCodeApplicationContext;
+        this.owner = owner;
+        this.projectStateRegistry = projectStateRegistry;
+        this.workerThreadRegistry = workerThreadRegistry;
+        this.domainObjectCollectionFactory = domainObjectCollectionFactory;
+        this.calculatedValueContainerFactory = calculatedValueContainerFactory;
+    }
+
+    /**
+     * Creates a new {@link DefaultConfiguration} instance.
+     */
+    DefaultConfiguration create(
+        String name,
+        ConfigurationsProvider configurationsProvider,
+        Factory<ResolutionStrategyInternal> resolutionStrategyFactory,
+        RootComponentMetadataBuilder rootComponentMetadataBuilder
+    ) {
+        ListenerBroadcast<DependencyResolutionListener> dependencyResolutionListeners =
```

Comment:
Why is this stored as a local instead of created inline in the `newInstance` call?

Reply:
Readability and potentially better stack traces. Personally, I don't care, but this seemed to read better IMHO.